### PR TITLE
WIP: Refactor claim index [currently being investigated, dont merge]

### PR DESCRIPTION
**Pull request by shyba**

### CODE REVIEW

File: src/rpc/claimtrie.cpp [hunk 1]
```diff
@@ -398,17 +398,16 @@ UniValue getclaimbyid(const UniValue& params, bool fHelp)
     LOCK(cs_main);
     uint160 claimId = ParseClaimtrieId(params[0], "Claim-id (parameter 1)");
     UniValue claim(UniValue::VOBJ);
-    std::string name;
     CClaimValue claimValue;
+    std::string name, sValue;
     pclaimTrie->getClaimById(claimId, name, claimValue);
-    if (claimValue.claimId == claimId)
+    if (pclaimTrie->getClaimById(claimId, name, claimValue) && getValueForClaim(claimValue.outPoint, sValue))
```
Comment (BrannonKing) [inline src/rpc/claimtrie.cpp]: `getClaimById` should not be called twice.

Comment (shyba) [reply src/rpc/claimtrie.cpp]: thanks, squashed and force pushed

File: src/rpc/claimtrie.cpp [hunk 2]
```diff
@@ -398,17 +398,15 @@ UniValue getclaimbyid(const UniValue& params, bool fHelp)
...
     {
         std::vector<CSupportValue> supports;
         CAmount effectiveAmount = pclaimTrie->getEffectiveAmountForClaimWithSupports(
             name, claimValue.claimId, supports);
 
-        std::string sValue;
-        getValueForClaim(claimValue.outPoint, sValue);
+        claim.push_back(Pair("value", sValue));
         claim.push_back(Pair("name", name));
         claim.push_back(Pair("value", sValue));
```
Comment (lbrynaut) [inline src/rpc/claimtrie.cpp]: Duplicate of line 409 (remove either this or your addition)

File: src/test/claimtriebranching_tests.cpp [hunk 3]
```diff
@@ -831,7 +831,7 @@ BOOST_AUTO_TEST_CASE(get_claim_by_id_test)
 {
     ClaimTrieChainFixture fixture;
     const std::string name = "test";
-    CMutableTransaction tx1 = fixture.MakeClaim(fixture.GetCoinbase(), name, "one", 1);
+    CMutableTransaction tx1 = fixture.MakeClaim(fixture.GetCoinbase(), name, "one", 2);
```
Comment (lbrynaut) [inline src/test/claimtriebranching_tests.cpp]: Why is this changing?

File: src/test/claimtriebranching_tests.cpp [hunk 4]
```diff
@@ -844,15 +844,28 @@ BOOST_AUTO_TEST_CASE(get_claim_by_id_test)
     fixture.MakeSupport(fixture.GetCoinbase(), tx1, name, 4);
     fixture.IncrementBlocks(1);
 
-    CMutableTransaction tx2 = fixture.MakeClaim(fixture.GetCoinbase(), name, "two", 1);
+    CMutableTransaction tx2 = fixture.MakeClaim(fixture.GetCoinbase(), name, "two", 2);
```
Comment (lbrynaut) [inline src/test/claimtriebranching_tests.cpp]: Why is this changing?

File: src/test/claimtriebranching_tests.cpp (hunk 3 above)
Comment (shyba) [reply src/test/claimtriebranching_tests.cpp]: So the update can test the value stored at the index also gets updated from 2 to 1.

File: src/test/claimtriebranching_tests.cpp (hunk 4 above)
Comment (shyba) [reply src/test/claimtriebranching_tests.cpp]: typo, fixing. I meant to do this only to tx1

**Comment by shyba** [comment]
squashed, fixed style, duplicated json field and typo on tests.

**Comment by shyba** [comment]
Added a command to debug the issue:
```
./lbrycrd-cli help getmissingclaimsfromindex                                            
getmissingclaimsfromindex
Return the claims that arent on claims_by_id index.
Arguments:
None
Result:
[
  "name"           (string) the claimed name
  "claimId"        (string) the claim id
]
```
Please don't merge that commit. We can remove it once we have a definitive fix or the reason for that to be happening.

**Comment by lbrynaut** [comment]
@shyba Thanks for bringing this issue up.  I looked into it a little bit and have a theory on what's going wrong, but will need some more time analyzing before testing a fix.  It's unrelated to this though, so no need to merge here.  If there's an existing issue related to this, please link it here.

**Comment by shyba** [comment]
That one is actually a cosmetic byproduct from when I was investigating. It's some unit tests and a refactor. The only issue being solve in this PR is when the `value` field fails it won't return a result instead of returning an empty value. However that's also caused by the index issues.
If the refactor and the unit tests proves to be useful you can merge them, but otherwise we can close this PR. It's mostly investigation byproducts.

**Comment by lbrynaut** [comment]
> If the refactor and the unit tests proves to be useful you can merge them, but otherwise we can close this PR. It's mostly investigation byproducts.

See #197.  If/when that's merged, please issue a PR to re-add the unit testing.

**Pull request closed without merging**